this now makes it not hang any more, but it does not work. solutions suggested?

1. Subprocess isolation — spawn CrashReporter.report() in a separate JVM process (its own launch, no -XstartOnFirstThread/headless), fully decoupled from the game. Keeps the full log-attaching Swing UI working; moderate effort (new process-launch plumbing, passing the throwable/log path across the process boundary).

2. Bypass Swing entirely — replace the reporter action with a ProcessBuilder-based "open the GitHub issues page + reveal the log folder" action, no AWT at all. Much less code, but loses the reporter's built-in log-attaching/formatting UI.

3. Stop here for now — leave the crash-reporter dialog non-functional on macOS under this launch config; the earlier background-thread fixes already prevent it from freezing the whole game, which may be enough for the moment.

```

  2026-07-27 00:56:58.179 java[7021:11519990] [JRSAppKitAWT markAppIsDaemon]: Process manager already initialized: can't fully enable headless mode.
  java.lang.reflect.InvocationTargetException
          at java.desktop/java.awt.EventQueue.invokeAndWait(EventQueue.java:1303)
          at java.desktop/java.awt.EventQueue.invokeAndWait(EventQueue.java:1278)
          at java.desktop/javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1464)
          at org.terasology.crashreporter.CrashReporter.report(CrashReporter.java:53)
          at org.terasology.engine.rendering.nui.layers.mainMenu.ExtrasMenuScreen.lambda$initialise$5(ExtrasMenuScreen.java:50)
          at java.base/java.lang.Thread.run(Thread.java:1516)
  Caused by: java.awt.HeadlessException
          at java.desktop/java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:158)
          at java.desktop/java.awt.Window.init(Window.java:471)
          at java.desktop/java.awt.Window.<init>(Window.java:423)
          at java.desktop/java.awt.Window.<init>(Window.java:563)
          at java.desktop/java.awt.Dialog.<init>(Dialog.java:647)
          at java.desktop/java.awt.Dialog.<init>(Dialog.java:496)
          at java.desktop/javax.swing.JDialog.<init>(JDialog.java:426)
          at org.terasology.crashreporter.CrashReporter.showModalDialog(CrashReporter.java:91)
          at org.terasology.crashreporter.CrashReporter$1.run(CrashReporter.java:64)
          at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
          at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:714)
          at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:693)
          at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
          at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
          at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
          at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
          at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
          at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

--- a/engine/src/main/java/org/terasology/engine/core/modes/StateLoading.java
+++ b/engine/src/main/java/org/terasology/engine/core/modes/StateLoading.java
@@ -247,7 +247,7 @@ public class StateLoading implements GameState {
                 String errorMessage = String.format("Failed to load game. There was an error during \"%s\".",
                         current == null ? "the last part" : current.getMessage());
                 gameEngine.changeState(new StateMainMenu(errorMessage));
-                CrashReporter.report(e, LoggingContext.getLoggingPath());
+                new Thread(() -> CrashReporter.report(e, LoggingContext.getLoggingPath()), "CrashReporter").start();
                 return;
             }
         }

--- a/engine/src/main/java/org/terasology/engine/rendering/nui/layers/ingame/ExtraMenuScreen.java
+++ b/engine/src/main/java/org/terasology/engine/rendering/nui/layers/ingame/ExtraMenuScreen.java
@@ -32,8 +32,9 @@ public class ExtraMenuScreen extends CoreScreenLayer {
 
         WidgetUtil.trySubscribe(this, "telemetry", button -> triggerForwardAnimation(TelemetryScreen.ASSET_URI));
         WidgetUtil.trySubscribe(this, "devTools", widget -> triggerForwardAnimation(DevToolsMenuScreen.ASSET_URI));
-        WidgetUtil.trySubscribe(this, "crashReporter", widget -> CrashReporter.report(new Throwable("There is no error."),
-                LoggingContext.getLoggingPath(), CrashReporter.MODE.ISSUE_REPORTER));
+        WidgetUtil.trySubscribe(this, "crashReporter", widget -> new Thread(() ->
+                CrashReporter.report(new Throwable("There is no error."),
+                        LoggingContext.getLoggingPath(), CrashReporter.MODE.ISSUE_REPORTER), "CrashReporter").start());
         WidgetUtil.trySubscribe(this, "close", widget -> triggerBackAnimation());
     }
 

--- a/facades/PC/src/main/java/org/terasology/engine/Terasology.java
+++ b/facades/PC/src/main/java/org/terasology/engine/Terasology.java
@@ -323,7 +323,7 @@ public final class Terasology implements Callable<Integer> {
         Path logPath = LoggingContext.getLoggingPath();
 
         if (!GraphicsEnvironment.isHeadless() && crashReportEnabled) {
-            CrashReporter.report(throwable, logPath);
+            new Thread(() -> CrashReporter.report(throwable, logPath), "CrashReporter").start();
         } else {
             throwable.printStackTrace();
             System.err.println("For more details, see the log files in " + logPath.toAbsolutePath().normalize());

